### DISCUSSION TITLE
Is it possible to extend just picker builder to implement custom UI?

### DISCUSSION THREAD

**Discussion by VladanRandjelovic** [fluttercandies/flutter_wechat_assets_picker#158] [Q&A] [answered]
Is It possible to extend just picker builder and use existing picker just with custom UI, if possible i would not use Navigator and routing in my UI...

Thank You in advance

**Comment by AlexV525** [comment]
I've reconsidered the part of overriding the delegate, maybe there can be a singleton or a static member for `AssetPicker` like `AssetPicker.delegate` for a permanently override:

```dart
class AssetPicker<A, P> {
  //...
  static AssetPickerBuilderDelegate? delegate;
}
```

Or we can split methods into two pick methods:

* AssetPicker.pickAsset(...)
* AssetPicker.pickAssetWithDelegate(...)

@VladanRandjelovic Would this implementation fit with your case?

/cc @yanivshaked

**Reply by VladanRandjelovic** [reply]
I think second one would be great, so we could actually do just the custom builder delegate and use pick method as it is (and just pass the delegate), which btw I think is great. And one more thing, thanks for the quick response mate, keep up the good work!

**Reply by AlexV525** [reply]
This will be updated with #108 if possible.

**Reply by VladanRandjelovic** [reply]
Cool. Thanks again!

**Comment by AlexV525** [comment, marked as answer]
It's been release with `6.0.0-dev.1` version.

**Reply by VladanRandjelovic** [reply]
cool thanks mate! :)

**Reply by yanivshaked** [reply]
Use `6.0.0-dev.2` and above (it contains a fix related to this feature)

**Reply by VladanRandjelovic** [reply]
it's still a must to use Navigator push, or the navigator is only used for pushing the data to the UI? I'm not sure i got that part right

**Reply by AlexV525** [reply]
> Is It possible to extend just picker builder and use existing picker just with custom UI

Previously you asked about minimized delegate extends and usage, so I guess new changes are perfect for this request.

> if possible i would not use Navigator and routing in my UI

Internal navigator usage is required and not exempt for that.

**Reply by VladanRandjelovic** [reply]
ok, now it's much clearer, thanks again mate.
